build-system/common/esbuild-babel.js

```diff
@@ -99,12 +115,89 @@ function getEsbuildBabelPlugin(
             rehash,
             getFileBabelOptions(babelOptions, filename)
           );
-          babelMaps?.set(filename, transformed.map);
+
+          babelMaps.set(filename, transformed.map);
+          postLoad?.();
           return {contents: transformed.code};
         }
       );
+
+      build.onEnd(async (result) => {
+        const {outputFiles} = result;
+        const code = outputFiles.find(({path}) => !path.endsWith('.map'));
+        const map = outputFiles.find(({path}) => path.endsWith('.map'));
+
+        if (!map) {
+          return;
+        }
+
+        const root = path.dirname(map.path);
+        const remapped = remapping(
+          map.text,
+          (f, ctx) => {
+            // The Babel tranformed file and the original file have the same
+            // path, which makes it difficult to distinguish during remapping's
+            // load phase. To prevent an infinite recursion, we check if the
+            // importer is ourselves (which is nonsensical) and early exit.
+            if (f === ctx.importer) {
+              return null;
+            }
+
+            const file = path.join(root, f);
+            const map = babelMaps.get(file);
+            if (!map) {
+              if (file.includes('/node_modules/')) {
+                // Excuse node_modules since they may have been marked external
+                // (and so not processed by babel).
+                return null;
+              }
+              throw new Error(`failed to find sourcemap for babel file "${f}"`);
+            }
+            return map;
+          },
+          !argv.full_sourcemaps
+        );
+
+        const sourcemapJson = remapped.toString();
+        replaceOutputFile(outputFiles, map, sourcemapJson);
+        if (inlineSourcemap) {
+          const base64 = Buffer.from(sourcemapJson).toString('base64');
+          replaceOutputFile(
+            outputFiles,
+            code,
+            code.text.replace(
+              /sourceMappingURL=.*/,
+              `sourceMappingURL=data:application/json;charset=utf-8;base64,${base64}`
+            )
+          );
+        }
+      });
+    },
+  };
+}
+
+/**
+ * @param {Array<import('esbuild').OutputFile>} outputFiles
+ * @param {import('esbuild').OutputFile} original
+ * @param {string} text
+ */
+function replaceOutputFile(outputFiles, original, text) {
+  const index = outputFiles.indexOf(original);
+  if (index === -1) {
+    throw new Error(`couldn't find outputFile ${original.path}`);
+  }
+
+  let contents;
+  const file = {
+    path: original.path,
+    text,
+    get contents() {
+      // eslint-disable-next-line local/no-forbidden-terms
```

Comment:
Nit: add this as an exception in build-system/test-configs/forbidden-terms.js instead of an explicit eslint-disable here?